aci/resource_aci_fvepgselector.go

```diff
@@ -0,0 +1,276 @@
+	d.Set("description", fvEPgSelector.Description)
+	fvEPgSelectorMap, err := fvEPgSelector.ToMap()
+	if err != nil {
+		return d, err
+	}
+	d.Set("annotation", fvEPgSelectorMap["annotation"])
+	d.Set("match_epg_dn", fvEPgSelectorMap["matchEpgDn"])
+	d.Set("name", fvEPgSelectorMap["name"])
+	d.Set("name_alias", fvEPgSelectorMap["nameAlias"])
+	return d, nil
+}
+
+func resourceAciEndpointSecurityGroupEPgSelectorImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	fvEPgSelector, err := getRemoteEndpointSecurityGroupEPgSelector(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setEndpointSecurityGroupEPgSelectorAttributes(fvEPgSelector, d)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciEndpointSecurityGroupEPgSelectorCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] EndpointSecurityGroupEPgSelector: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+	matchEpgDn := d.Get("match_epg_dn").(string)
+	EndpointSecurityGroupDn := d.Get("endpoint_security_group_dn").(string)
+
+	fvEPgSelectorAttr := models.EndpointSecurityGroupEPgSelectorAttributes{}
+	nameAlias := ""
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		nameAlias = NameAlias.(string)
+	}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		fvEPgSelectorAttr.Annotation = Annotation.(string)
+	} else {
+		fvEPgSelectorAttr.Annotation = "{}"
+	}
+
+	if MatchEpgDn, ok := d.GetOk("match_epg_dn"); ok {
+		fvEPgSelectorAttr.MatchEpgDn = MatchEpgDn.(string)
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		fvEPgSelectorAttr.Name = Name.(string)
+	}
+	fvEPgSelector := models.NewEndpointSecurityGroupEPgSelector(fmt.Sprintf("epgselector-[%s]", matchEpgDn), EndpointSecurityGroupDn, desc, nameAlias, fvEPgSelectorAttr)
+
+	err := aciClient.Save(fvEPgSelector)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Partial(true)
+	//d.SetPartial("name")
+	d.Partial(false)
+	checkDns := make([]string, 0, 1)
+
+	if relationTofvRsMatchEPg, ok := d.GetOk("relation_fv_rs_match_e_pg"); ok {
+		relationParam := relationTofvRsMatchEPg.(string)
+		checkDns = append(checkDns, relationParam)
+
+	}
+
+	d.Partial(true)
+	err = checkTDn(aciClient, checkDns)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Partial(false)
+
+	if relationTofvRsMatchEPg, ok := d.GetOk("relation_fv_rs_match_e_pg"); ok {
+		relationParam := relationTofvRsMatchEPg.(string)
+		err = aciClient.CreateRelationfvRsMatchEPg(fvEPgSelector.DistinguishedName, fvEPgSelectorAttr.Annotation, relationParam)
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.Partial(true)
+		//d.SetPartial("relation_fv_rs_match_e_pg")
+		d.Partial(false)
```

Comment:
Do we still need those Partial if we don't do anything in between?

Reply:
No. I'll remove it